### PR TITLE
Add GetUidNumber function

### PR DESCRIPTION
**Pull request by limed**
This returns the uidNumber attribute from LDAP

### CODE REVIEW

File: mozldap.go [hunk 1]
```diff
@@ -296,6 +296,36 @@ func (cli *Client) GetUserDNById(uid string) (dn string, err error) {
 	return
 }
 
+// GetUserUidNumber returns the UID number of a user using a shortdn
+//
+// example: cli.GetUserUidNumber("mail=jvehent@mozilla.com")
+func (cli *Client) GetUserUidNumber(shortdn string) (uidNumber string, err error) {
```
Comment (jvehent) [inline mozldap.go]: It's a bit odd that `GetUserUidNumber` would return a string and not an int. Would it make sense to return the uidNumber as an int? It's pretty easy to do with

``` go
uidNumber, err = strconv.Atoi(val)
if err != nil {
panic(err)
}
```

Comment (limed) [inline mozldap.go]: That sounds good to me, I'll revise patch reflect this

File: mozldap.go [hunk 2]
```diff
@@ -296,6 +296,39 @@ func (cli *Client) GetUserDNById(uid string) (dn string, err error) {
...
+		panic(err)
+	}
+	for _, entry := range entries {
+		for _, attr := range entry.Attributes {
+			if attr.Name != "uidNumber" {
+				continue
+			}
+			for _, val := range attr.Values {
+				uidNumber, err = strconv.ParseUint(val, 10, 64)
+				if err != nill {
```
Comment (jvehent) [inline mozldap.go]: `nill` ? you mean `nil`

Comment (limed) [inline mozldap.go]: Opps yes :)

**Pull request merged**